### PR TITLE
nsh: Improve performance of `help` by line buffer

### PR DESCRIPTION
**Pull request by no1wudi**
## Summary
This change reduce the usage of `printf`,
which will improve both CPU usage and IO performance.
## Impact
help command
## Testing
esp32c3 and telnet

### CODE REVIEW

**Comment by no1wudi** [comment]
@masayuki2009 This PR improve the performance of help command without NET_TCP_WRITE_BUFFER.
BTW, maybe the `ps` command meet the same issue.

https://user-images.githubusercontent.com/9030616/222391336-2057d122-abf1-432d-8550-ae0313d1d0cb.mp4

**Comment by masayuki2009** [comment]
@no1wudi 

I confirmed that this PR fixes the help command issue.

>BTW, maybe the ps command meet the same issue.

I think so too. 
Can you add a new commit to this PR?

**Comment by no1wudi** [comment]
> I think so too. Can you add a new commit to this PR?

Yes, Ill' try to optimize it by merge multiple nsh_output into one.

File: nshlib/nsh_command.c [hunk 1]
```diff
@@ -650,22 +652,32 @@ static inline void help_cmdlist(FAR struct nsh_vtbl_s *vtbl)
...
+
       for (j = 0, k = i;
            j < cmdsperline && k < NUM_CMDS;
            j++, k += ncmdrows)
         {
-          nsh_output(vtbl, "%s", g_cmdmap[k].cmd);
+          /* Copy the cmd name to line buffer */
+
+          offset += strlcpy(line + offset, g_cmdmap[k].cmd,
+                    sizeof(line) - offset);
```
Comment (xiaoxiang781216) [inline nshlib/nsh_command.c:667]:
```suggestion
                           sizeof(line) - offset);
```

**Comment by no1wudi** [comment]
@masayuki2009 
Before commit 2:

https://user-images.githubusercontent.com/9030616/222459070-a819004c-1049-49e3-bc8c-db5a41c38bd4.mp4

After:


https://user-images.githubusercontent.com/9030616/222459117-523a8423-c1ec-4b73-8c99-6fe008416af4.mp4

**Review by xiaoxiang781216 — APPROVED** [review]

**Review by jerpelea — APPROVED** [review]

**Review by pkarashchenko — APPROVED** [review]

**Comment by masayuki2009** [comment]
```
====================================================================================
Configuration/Tool: sim/vpnkit
------------------------------------------------------------------------------------
  Cleaning...
  Configuring...
  Building NuttX...
In file included from nsh_proccmds.c:40:
nsh_proccmds.c: In function 'ps_callback':
Error: nsh_proccmds.c:522:16: error: format '%lu' expects a matching 'long unsigned int' argument [-Werror=format=]
  522 |                "%06lu "
      |                ^~~~~~~~
nsh_console.h:53:49: note: in definition of macro 'nsh_output'
   53 | # define nsh_output(v, ...)    (v)->output(v, ##__VA_ARGS__)
      |                                                 ^~~~~~~~~~~
nsh_proccmds.c:523:21: note: format string is defined here
  523 |                "%06lu "
      |                 ~~~~^
      |                     |
      |                     long unsigned int
cc1: all warnings being treated as errors
```

**Review by xiaoxiang781216 — APPROVED** [review]

**Comment by tmedicci** [comment]
Hi @no1wudi , could you please take a look on https://github.com/apache/nuttx-apps/issues/1620?

**Comment by no1wudi** [comment]
> Hi @no1wudi , could you please take a look at https://github.com/apache/nuttx-apps/issues/1620?

Yes, I will take look，does ps command meet the same issue？

**Pull request merged**